File: 3120-count-the-number-of-special-characters-i/3120-count-the-number-of-special-characters-i.py
class Solution:
    def numberOfSpecialChars(self, word: str) -> int:

        st = set()

        for ch in word:
            st.add(ch)

        count = 0

        for ch in word:

            if ch.islower():

                upper = ch.upper()

                if upper in st:
                    count += 1

                    st.remove(upper)

        return count

File: 3120-count-the-number-of-special-characters-i/3120-count-the-number-of-special-characters-i.java
class Solution {
    public int numberOfSpecialChars(String word) {
        HashSet<Character> set=new HashSet<>();
        for(int i=0; i<word.length(); i++){
            set.add(word.charAt(i));
        }
        int count=0;
        for(int i=0; i<word.length(); i++){
            char ch=word.charAt(i);
            if(ch-'a'>=0){
                char c=(char)('A'+(ch-'a'));
                if(set.contains(c)){
                    count++;
                    set.remove(c);
                }
            }
        }
        return count;
    }
}